daemon/daemon.go

```diff
@@ -265,19 +265,19 @@ func getTrustedAccountKey() string {
 }
 
 // New Daemon
-func New() *Daemon {
+func New() (*Daemon, error) {
 	ovld, err := overlord.New()
 	if err != nil {
-		panic(err.Error())
+		return nil, err
 	}
 	db, err := asserts.OpenSysDatabase(getTrustedAccountKey())
 	if err != nil {
-		panic(err.Error())
+		return nil, err
 	}
 	interfacesRepo := interfaces.NewRepository()
 	for _, iface := range builtin.Interfaces() {
 		if err := interfacesRepo.AddInterface(iface); err != nil {
-			panic(err.Error())
+			return nil, err
```

Comment:
isn't there cleanup to be done here? Of the `*Database` and `*Overlord` you now have I mean.